muikku-core-plugins/src/main/resources/META-INF/resources/scripts/gui/communicator.js

```diff
@@ -82,29 +82,33 @@ $(document).ready(function(){
       var mCont = $(CommunicatorImpl.msgContainer);
       this._addLoading(CommunicatorImpl.msgContainer);
       
-      mApi().communicator.messages.read(communicatorMessageId).callback($.proxy(function (err, result) {
-        for (var i = 0; i < result.length; i++) {
-          result[i].caption = $('<div>').html(result[i].caption).text();
-          result[i].content = $('<div>').html(result[i].content).text();
-          
-          var sId = result[i].id;
-          mApi().communicator.communicatormessages.sender.read(sId)
-            .callback(function (err, user) {  
+      mApi().communicator.messages.read(communicatorMessageId).callback($.proxy(function (err, messages) {
+        
+        // TODO: Recipients issue no: #875 
+        
+        $.each(messages, function (index, message){
+           messages[index].caption = $('<div>').html(message.caption).text();
+           messages[index].content = $('<div>').html(message.content).text();
+           
+           mApi().communicator.communicatormessages.sender.read(message.id)
+           .callback(function (err, user) {  
```

Comment:
Rest call inside a loop will not work as expected and can lead to serious performance issues